lib/test_lib.c

```diff
@@ -0,0 +1,35 @@
+#include "../src/extern_library_helper.h"
+#include <stdio.h>
+
+#define NUM_FUNCTIONS 1
+LibraryDefinition expose_library() {
+    char* names[NUM_FUNCTIONS] = {"day_of_week"};
+    LibraryDefinition lib = {NUM_FUNCTIONS, names};
```

Comment:
I don't like that the user needs to keep track of the number of functions it implemented.

Maybe, instead of a counter, we could have a blank last entry, like this.

```c
    LibraryDefinition lib = {
        "day_of_week",
        0,
    }
```

Reply:
I created a macro in which the user only needs to provide the name of the functions, like this:
```c
DEFINE_LIBRARY("day_of_week", "hello", "foo")
```

Reply:
I think the `FUNCTION` is not needed, it just declares a struct

Reply:
It's not necessary, but I think it makes it more legible. Here's an example of what it would look like without this macro:
```c
DEFINE_LIBRARY(
    (extern_function){"day_of_week", (function_pointer) day_of_week}
);
```
What do you think?

Reply:
The casting is necessary? Also, don't declare the structs with typedef.

Reply:
Now the user no longer has to cast (the macro does that), I've removed the typedef from the structs and I've also simplified the code by removing the function counter.